pkg/auth/oauth/registry/expirationvalidator.go

```diff
@@ -0,0 +1,28 @@
+package registry
+
+import (
+	"errors"
+	"time"
+
+	"github.com/openshift/origin/pkg/auth/authenticator"
+	"github.com/openshift/origin/pkg/oauth/apis/oauth"
+	"github.com/openshift/origin/pkg/user/apis/user"
+)
+
+var errExpired = errors.New("token is expired")
+
+func NewExpirationValidator() authenticator.OAuthTokenUserValidator {
+	return authenticator.OAuthTokenUserValidatorFunc(
+		func(token *oauth.OAuthAccessToken, _ *user.User) error {
+			if token.ExpiresIn > 0 {
+				if token.CreationTimestamp.Time.Add(time.Duration(token.ExpiresIn) * time.Second).Before(time.Now()) {
```

Comment:
It be more readbale if the conversion of token.ExpiresIn to a duration would be doen via a helper function